src/vs/workbench/parts/debug/browser/exceptionWidget.ts

```diff
@@ -33,15 +33,40 @@ export class ExceptionWidget extends ZoneWidget {
 		const fontInfo = this.editor.getConfiguration().fontInfo;
 		this.container.style.fontSize = `${fontInfo.fontSize}px`;
 		this.container.style.lineHeight = `${fontInfo.lineHeight}px`;
-
-		let title = $('.title');
-		title.textContent = nls.localize('exceptionThrown', 'Exception occurred');
-		dom.append(container, title);
-
 		const thread = this.debugService.getViewModel().focusedThread;
+
 		if (thread && thread.stoppedDetails) {
+			let title = $('.title');
 			let msg = $('.message');
-			msg.textContent = thread.stoppedDetails.text;
+
+			if (this.exceptionInfo) {
+				let conditionMessage;
+				switch (this.exceptionInfo.body.breakMode) {
+					case 'never':
+						conditionMessage = nls.localize('neverException', 'User-handled exception has occurred.');
+						break;
+					case 'always':
+						conditionMessage = nls.localize('alwaysException', 'Always-breaking exception has occurred.');
+						break;
+					case 'unhandled':
+						conditionMessage = nls.localize('unhandledException', 'Unhandled exception has occurred.');
+						break;
+					case 'userUnhandled':
+						conditionMessage = nls.localize('userUnhandledException', 'User-unhandled exception has occurred.');
+						break;
+					default:
+						conditionMessage = '';
+						break;
+				}
+
+				title.textContent = `${conditionMessage} ${this.exceptionInfo.body.description}`;
```

Comment:
That's how it looks now:

![image](https://cloud.githubusercontent.com/assets/2239563/24190317/a9a1a3fe-0ee8-11e7-8425-2313cf0d8351.png)